core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -141,7 +141,7 @@ private[spark] class HighlyCompressedMapStatus private (
     private[this] var numNonEmptyBlocks: Int,
     private[this] var emptyBlocks: RoaringBitmap,
     private[this] var avgSize: Long,
-    @transient private var hugeBlockSizes: Map[Int, Byte])
+    private[this] var hugeBlockSizes: Map[Int, Byte])
```

Comment:
actually we can do better: use a bitmap to track which block id has size info, and a byte array to store these size data. so the format can be: `[num blocks] [bit map], [size array]`